report/11_Systems-Design.md

```diff
@@ -1 +1,11 @@
-Design of your ITU-MiniTwit systems
+A component digram describing the flow of data through the MiniTwit system itself can be seen in the figure below.  
+
+![Component diagram](./images/component_diagram.png)
+
+The MiniTwit application has two main C# Projects, MiniTwit.Web.App, containing the web app, and MiniTwit.Models containing the repositories for the different types of data handled by the application. The application also has other projects containing the MiniTwitContext connecting to the Database, or containing utility functions, but these are not illustrated in the diagram above. 
+
+The ```MinitwitContext``` (based in EntityFrameworkCore) functions as the database abstraction layer, and allows the repositories to access their relevant data. The repositories then contain the relevant methods for passing the data on in specific formats to the frontend controllers, contained in ```MiniTwit.Web.App```. The ```HomeController``` passes this data on to the views in the Views directory, where they are used when accessed in a browser, while the ```APIcontroller``` instead exposes the API that the simulator uses.
```

Comment:
> (based in EntityFrameworkCore)

reword this  part. 

"repositories" if the design pattern is meant here, it has to be explained. 

HomeController and Views directory makes no sense if it has not been mentioned that it is a Asp.net application yet.